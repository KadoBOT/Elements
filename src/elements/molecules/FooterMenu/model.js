export type FooterMenu = [string]

export type State = FooterMenu[]
