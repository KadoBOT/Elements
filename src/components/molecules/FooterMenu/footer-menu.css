.footer-menu {
  display: flex;
  align-items: center
}

.footer-menu__item {
  margin: 0px 30px;
}

.footer-menu__item * {
  font-size: 18px;
}

.footer-menu__item:last-child {
  margin: 0px 0px;
}
