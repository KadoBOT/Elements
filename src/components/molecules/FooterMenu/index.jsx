import React, {PropTypes} from 'react'
import R from 'ramda'

import { MenuItem } from '../../atoms'
import './footer-menu.css'

const FooterMenu = ({menuItems}) => {
  return(
    <div className="footer-menu">
      {R.map(m =>
        <div className="footer-menu__item">
          <MenuItem item={m.toUpperCase()} />
        </div>
      ,menuItems)}
    </div>
  )
}

FooterMenu.propTypes = {
  props: PropTypes.type
}

export default FooterMenu
