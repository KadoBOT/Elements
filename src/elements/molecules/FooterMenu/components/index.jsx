import React, {PropTypes} from 'react'
import { connect } from 'react-redux'
import R from 'ramda'

import { MenuItem } from '../../atoms'
import './footer-menu.css'

const FooterMenu = ({menuItems}) => (
  <div className="footer-menu">
    {R.map(m =>
      <div key={m} className="footer-menu__item">
        <MenuItem item={m.toUpperCase()} />
      </div>
    ,menuItems)}
  </div>
)

FooterMenu.propTypes = {
  menuItems: PropTypes.array.isRequired,
}

const mapStateToProps = state => ({
  menuItems: state.menuItems
})

export default connect(mapStateToProps)(FooterMenu)
