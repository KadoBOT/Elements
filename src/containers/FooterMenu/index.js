import { connect } from 'react-redux'

import { FooterMenu } from '../../components/molecules'

const mapStateToProps = state => ({
  menuItems: state.menuItems
})

export default connect(mapStateToProps)(FooterMenu)
