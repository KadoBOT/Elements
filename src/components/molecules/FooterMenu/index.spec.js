import React from 'react'
import { shallow, render, mount } from 'enzyme';

import FooterMenu from './'

it('should render the FooterMenu', () => {
    const wrapper = shallow(
        <FooterMenu menuItems={[]} />
    );
    expect(wrapper).toMatchSnapshot();
});
