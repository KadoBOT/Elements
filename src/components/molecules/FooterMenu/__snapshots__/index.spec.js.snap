exports[`test should render the FooterMenu 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="footer-menu">
    
</div>,
  "nodes": Array [
    <div
      className="footer-menu">
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FooterMenu
        menuItems={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "menuItems": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="footer-menu">
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="footer-menu">
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FooterMenu
    menuItems={Array []} />,
}
`;
